src/main/fc/fc_core.c

```diff
@@ -245,6 +250,13 @@ void updateArmingStatus(void)
             }
         }
 
+        if (IS_RC_MODE_ACTIVE(BOXGRAVEYARD) && graveyardModeAllowed) {
+            setArmingDisabled(ARMING_DISABLED_GRAVEYARD);
+            dispatchAdd(&preventModeChangesDispatchEntry, GRAVEYARD_PREVENT_MODE_CHANGES_DELAY);
+        } else if (!IS_RC_MODE_ACTIVE(BOXGRAVEYARD) && !graveyardModeAllowed) {
```

Comment:
I'd probably move the `else` part into `rc_modes` and make it work independently of the arming state - having the reset of the switch being dependent on being disarmed is not intuitive.

Reply:
Hmm, the reset of the switch is not dependent on being disarmed. It's about having the graveyard mode disabled at least once, so forgetting to flip the switch back when powering up the backup quad doesn't put the new quad into graveyard right away

Reply:
> Hmm, the reset of the switch is not dependent on being disarmed.

At the moment it is, because the check is only ever reached when disarmed. That is why I suggested changing it so that the reset isn't dependent on arming status.